client/extensions/woocommerce/app/store-stats/referrers/index.js

```diff
@@ -0,0 +1,99 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { find, sortBy } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import QuerySiteStats from 'components/data/query-site-stats';
+import { getSiteStatsNormalizedData } from 'state/stats/lists/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { getUnitPeriod } from 'woocommerce/app/store-stats/utils';
+import StoreStatsPeriodNav from 'woocommerce/app/store-stats/store-stats-period-nav';
+import JetpackColophon from 'components/jetpack-colophon';
+import Main from 'components/main';
+import Module from 'woocommerce/app/store-stats/store-stats-module';
+
+const STAT_TYPE = 'statsStoreReferrers';
+
+const Referrers = ( {
+	siteId,
+	query,
+	data,
+	selectedDate,
+	unit,
+	slug,
+	path,
+	querystring,
+	translate,
+} ) => {
+	const unitSelectedDate = getUnitPeriod( selectedDate, unit );
+	const selectedData = find( data, d => d.date === unitSelectedDate ) || { data: [] };
+	const sortedData = sortBy( selectedData.data, d => -d.sales );
```

Comment:
We do this in a few places now (grab a row by date, and then order by sales): Here, your stats widget, and my dashboard widget. Maybe we should have a helper function for referrer data? It could optionally take a `limit` param since we slice the results sometimes too.